src/main/java/bio/terra/common/DaoUtils.java

```diff
@@ -32,16 +33,24 @@ public static String orderByClause(EnumerateSortByParam sort, SqlSortDirection d
             .toString();
     }
 
-    public static void addFilterClause(String filter, MapSqlParameterSource params, List<String> clauses) {
+    public static void addFilterClause(String filter, MapSqlParameterSource params, List<String> clauses,
+                                              String datasetIdField) {
         if (!StringUtils.isEmpty(filter)) {
-            params.addValue("filter", DaoUtils.escapeFilter(filter));
-            clauses.add(" (name ILIKE :filter OR description ILIKE :filter) ");
+            GoogleRegion regionFilter = GoogleRegion.fromValue(filter);
+            if (regionFilter != null) {
+                params.addValue("region", regionFilter.name());
+                clauses.add(String.format(":region in (SELECT storage_resource.region FROM storage_resource" +
+                    " WHERE storage_resource.dataset_id = %s) ", datasetIdField));
```

Comment:
I don't think you can use the normal parameter sanitation methods here because `dataset_id` is a column reference and not a value. Using `addParam` will try to escape it as a value, but we need the raw column reference. The `datasetIdField` is never user-supplied, so I think its fine.

Reply:
Yes like Trevyn said, this is the name of the column which could either be `dataset.id` for the dataset enumerate endpoint or `snapshot_source.dataset_id` for snapshots

Reply:
Ahh, I got you.  Ok, I get it.  Do we typically use format?  Or do we want to use a template?  Just want to make sure we're consistent

Reply:
I could almost see this be a function that takes in the column name and returns the clause, but I don't feel super strongly about.  Just a thought